tests/test_warc_metadata_sidecar.py

```diff
@@ -76,6 +76,12 @@ def test_unknown_language():
     assert language is None
 
 
+def test_determine_sof404():
+    soft404_page = b'<h1>Page Not Found<h1>'
+    detected = sidecar.determine_soft404(soft404_page)
+    assert detected
```

Comment:
This seems like a broad assertion that could be more tailored. Perhaps `soft404.probability` should be patched, so you can `assert` you get the value set for the return value of the patched `soft404.probability`?